src/callback.jl

```diff
@@ -1,7 +1,13 @@
 """
-Register a function pointer as an R NativeSymbol. We technically are supposed to use
-R_registerRoutines. Starting from R 3.4, `R_MakeExternalPtrFn` is a part of R API in R 3.4.
-It is probably safe to such to make the external pointer.
+    makeNativeSymbolRef(fptr::Ptr{Cvoid})
+
+Register a function pointer as an R `NativeSymbol`.
+
+!!! note
+   `R_MakeExternalPtrFn` is not part of the official R API before
+    R 3.4 and we are supposed to use `R_registerRoutines` instead.
+    It is probably safe to such to make the external pointer, even
+    on older versions of R (although we don't officially support them.)
```

Comment:
I find this a bit difficult to follow as written. Is the following a correct understanding of the situation?

- R < 3.4
    - We don't support this R version
    - The functionality of `R_MakeExternalPtrFn` is exposed officially by `R_registerRoutines`
    - `R_MakeExternalPtrFn` does exist but is not part of R's official API
    - This Julia function might work with this version of R but we don't particularly care one way or another
- R ≥ 3.4
    - We support this version of R
    - `R_MakeExternalPtrFn` is part of R's official API
    - This Julia function is expected to work and it's a bug if it doesn't

If we don't support versions of R where the R function this Julia function wraps isn't officially supported by R then is it worth mentioning at all?

Reply:
correct. tbh, I had considered omitting it completely because the comment was largely historical